src/SumoCoders/FrameworkCoreBundle/BreadCrumb/BreadCrumbBuilder.php

```diff
@@ -0,0 +1,248 @@
+
+final class BreadCrumbBuilder
+{
+    /**
+     * @var bool
+     */
+    private $dontExtractFromTheRoute = false;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    /**
+     * @var FactoryInterface
+     */
+    private $factory;
+
+    /**
+     * @var array
+     */
+    private $items = array();
+
+    /**
+     * @param FactoryInterface         $factory
+     * @param EventDispatcherInterface $eventDispatcher
+     */
+    public function __construct(FactoryInterface $factory, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->factory = $factory;
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    /**
+     * Add items into the breadcrumb based on a given child.
+     *
+     * @param MenuItem $item
+     * @param string   $locale
+     */
+    protected function addItemsBasedOnTheChild(MenuItem $item, $locale)
+    {
+        if (null !== $item) {
+            $items = array();
+            $temporaryItem = $item;
+
+            while (null !== $temporaryItem->getParent()) {
+                $breadCrumb = new MenuItem($temporaryItem->getName(), $this->factory);
+                $breadCrumb->setLabel($temporaryItem->getLabel());
+
+                if ('#' !== $temporaryItem->getUri() && null !== $temporaryItem->getUri()) {
+                    $breadCrumb->setUri($temporaryItem->getUri());
+                }
+                $items[] = $breadCrumb;
+
+                $temporaryItem = $temporaryItem->getParent();
+            }
+
+            $home = new MenuItem('core.menu.home', $this->factory);
+            $home->setLabel('core.menu.home');
+            $home->setUri('/' . $locale);
+            $this->items[] = $home;
+
+            $this->items = array_merge($this->items, array_reverse($items));
+        }
+    }
+
+    /**
+     * Grab the menu
+     *
+     * This method will use the ConfigureMenuEvent to get all the items
+     *
+     * @return \Knp\Menu\ItemInterface
+     */
+    protected function getTheMenu()
+    {
+        $menu = $this->factory->createItem('root');
+
+        $this->eventDispatcher->dispatch(
+            ConfigureMenuEvent::EVENT_NAME,
+            new ConfigureMenuEvent(
+                $this->factory,
+                $menu
+            )
+        );
+
+        return $menu;
+    }
+
+    /**
+     * Find an item in the menu based on its URI
+     *
+     * @param MenuItem $menuItem
+     * @param          $uri
+     * @return MenuItem|null
+     */
+    protected function findItemBasedOnUri(MenuItem $menuItem, $uri)
+    {
+        if ($uri === $menuItem->getUri()) {
+            return $menuItem;
+        }
+
+        if (!$menuItem->hasChildren()) {
+            return null;
+        }
+
+        foreach ($menuItem->getChildren() as $child) {
+            $item = $this->findItemBasedOnUri(
+                $child,
+                $uri
+            );
+
+            if (null !== $item) {
```

Comment:
ik ben het eiglk ook wel met wouter eens
deze reactie geld ook voor alle andere plaatsen ;)
die assignment kan je ook alleen maar krijgen als je alleen een = schrijft in plaats van een == of in dit geval !== en daar dienen pull request ook al voor